### PR TITLE
Add usecases, applications, acceptance process, etc.

### PR DESCRIPTION
**Pull request by krgovind**
- Add more information about Goals and Non-Goals, specifically addressing no impact on Same Origin Policy.
- Add specific usecase examples.
- Add specific web platform applications related to browser privacy work.
- Update acceptance process/policy to propose a standard policy, verified by an independent entity.
- Address section on potential for technical enforcement of policy.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,103 +61,151 @@ then accepted by a browser if the set conforms to its policy.
...
+-   Issuing WebID [directed identifiers](https://github.com/WICG/WebID/blob/main/directed_identifiers.md) by First-Party Set, so the same account can be shared across multiple applications or services provided by the same first-party.
+-   Applying [Privacy Budget](https://github.com/bslassey/privacy-budget) across an entire First-Party Set, in order to prevent fingerprinting entropy from being accumulated across domains that are able to communicate in an unconstrained manner due to access to cross-domain, same-party cookies.
+-   Top and/or second level key for cache partitioning, potentially with site opt-in.
+
+This proposal is consistent with the same-origin policy. That is, Web Platform features must not use first-party sets to make one origin's state directly accessible to another origin in the set. For example, if a.example and b.example are in the same first-party set, the same-origin policy would still prevent `https://a.example` from accessing `https://b.example`'s cookies or IndexedDB databases.
```
Comment (torgo) [inline README.md:132]: This is confusing to me – however I think what is being left out is that there is a new class of cookies being considered - same party cookies - that **would** be able to be shared between a.example and b.example. Correct? If so, this should be called out here.

File: README.md [hunk 2]
```diff
@@ -58,103 +61,151 @@ then accepted by a browser if the set conforms to its policy.
...
+
+
+# Acceptance Process
+
+This section proposes a possible model for a First-Party Set acceptance process that could be shared across all browsers. However, many aspects of the process and policy will need to be tuned based on feedback from the web ecosystem.
+
+## Submission
+
+Sites will need to submit their proposed group of domains to a public tracker (such as a dedicated GitHub repository, like that of the [Public Suffix List](https://github.com/publicsuffix/list/wiki/Guidelines), and [Disconnect’s entities list](https://github.com/disconnectme/disconnect-tracking-protection/issues?q=is%3Aissue+%22entity%22+)), along with information needed to satisfy the UA policy. Technical verification of the submitter’s control over the domains may also require a challenge to be served at a `.well-known` location on each of the domains in the set.
```
Comment (rhiaro) [inline README.md:166]: Great to see expansion on this, but I'm still concerned about how this would work in practice. I acknowledge that it's (rightfully) still very open and subject to evolution. But I'm imagining either:

* there is one list that all UAs have come to agreement on policy on, and they all pull from the same list. I see the practicality of this, but it's dependent on UAs agreeing policy (not a guarantee) and for something with such a broad scope (any site on the web can play) feels far too much centralisation.
* sites have to submit their set for approval to *every* UA, which doesn't seem realistic. Even with just mainstream UAs right now, assuming they all implement FPS, I can see site admins submitting requests to MS, Apple, Mozilla and Google, but not bothering with Opera, Samsung, let alone Vivaldi, Brave, Tor... And what about mobile browsers? Whether or not requests need to be submitted to eg. Firefox mobile separately from Firefox desktop may not be obvious to site admins.
* the middle ground: some UAs agree policy and share a list, other UAs might fork the list, others might use the list but take liberties with how they implement it (eg. customise it). Besides fragmentation issues, this becomes confusing for site owners with regards to knowing where to submit their sets for verification.

Basically I'm just struggling to imagine how the verification process to determine whether sites are legitimately in the same set can reasonably scale to the whole web with this level of centralisation, and without [automation](https://github.com/privacycg/first-party-sets/issues/43).

**Review by rhiaro — COMMENTED** [review]
In general I think I'd like to see a lot more about what happens when a UA *doesn't* implement FPS when others do (including what happens to older browsers that are never going to get updated); and when different UAs ship with different FPS lists. How badly can things break from an end user pespective, and from a site owner/admin perspective?

The case I'm particularly worried about is sites blocking access to UAs that don't implement FPS (because they're old, or because they choose not to for privacy reasons) or users who have FPS turned off.

File: README.md (hunk 1 above)
Comment (krgovind) [reply README.md:132]: Even with the `SameParty` cookie attribute; cookies bound to a.example can never be accessed by b.example. The intention for that attribute is that b.example can access **it's own cookies** when embedded on a.example, assuming that a.example and b.example are within the same FPS. Developers can then opt-in to `SameParty` attributes even on browsers that don't yet block third-party cookies by default.

File: README.md [hunk 3]
```diff
@@ -58,103 +61,151 @@ then accepted by a browser if the set conforms to its policy.
...
+
+# Site-Declared Sets in Browsers
+
+Browsers should maintain a static list of site-declared groups of domains which meet UA ([User Agent](https://www.w3.org/WAI/UA/work/wiki/Definition_of_User_Agent)) policy, and ship it in the browser as a reliably updateable component. This is analogous to the list of [domains owned by the same entity](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/entities.json) used by Edge and Firefox to control cross-site tracking mitigations.
+
+The differences between this proposal and the use of the [Disconnect entities list](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/entities.json) in Edge and Firefox are:
+
+*   **All sites** with use-cases that depend on cross-domain, same-party communication will be required to declare one (or more) sets. As opposed to the Disconnect list, which only applies to sites [classified as a tracker](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/services.json).
```
[the remark this answers is not in the document]
Comment (krgovind) [reply README.md:140]: Oops, sorry, this is a typo/mis-statement. I meant to say that organizations that can divide up their domains into one or more sets. I'll fix it by removing the "or more" for now, and mention elsewhere that organizations can declare multiple disjoint sets (but that no domain can appear in more that one set).

Thanks for catching that!

File: README.md (hunk 2 above)
Comment (krgovind) [reply README.md:166]: Thanks for the feedback, Amy. Just acknowledging that I saw this. We're discussing the [technical-only enforcement issue](https://github.com/privacycg/first-party-sets/issues/43) that you linked to within PrivacyCG, and also plan to have another conversation on the policy aspects. I'll come back and address this once I hear from the others as well.

File: README.md (hunk 1 above)
Comment (torgo) [reply README.md:132]: It still feels t me like this needs to be better spelled out in the PR - and if you're going to talk about cookies here then it feels like you should also mention the SameParty cookie proposal just to clarify how that enables or does not enable crossing the privacy boundary.

Comment (krgovind) [reply README.md:132]: Thanks for the feedback, Dan! I'll iterate on this in a separate PR (created #47 to track).

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (krgovind) [reply README.md]: Fixed in latest commit.

File: README.md (hunk 2 above)
Comment (krgovind) [reply README.md:166]: Amy, FYI, I opened issue #48 to keep track of this feedback.

**Comment by krgovind** [comment]
@torgo @rhiaro - thank you for your feedback! FYI, I opened #47 and #48 to keep track of your feedback that I plan to resolve separately.

For #47 , we are working on additional text/visual aids to help explain the application of FPS to cookies (via the `SameParty` attribute) better, especially within the context of how this conforms to Same Origin Policy.

For #48 , we have upcoming cross-browser discussion on UA Policy happening in the PrivacyCG, where we hope to make progress on the concerns that Amy raised.

For now, I plan to merge this PR; but please feel free to open new issues if you have any additional comments!
